instrumentation/servlet/servlet-5.0/library/src/main/java/io/opentelemetry/instrumentation/servlet/jakarta/v5_0/JakartaServletHttpServerTracer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.servlet.jakarta.v5_0;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.instrumentation.api.servlet.ServletSpanNaming;
+import io.opentelemetry.instrumentation.api.tracer.ServerSpan;
+import io.opentelemetry.instrumentation.servlet.MappingResolver;
+import io.opentelemetry.instrumentation.servlet.ServletHttpServerTracer;
+import jakarta.servlet.Servlet;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.ServletRegistration;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.Collection;
+
+public class JakartaServletHttpServerTracer
+    extends ServletHttpServerTracer<HttpServletRequest, HttpServletResponse> {
+  private static final JakartaServletHttpServerTracer TRACER = new JakartaServletHttpServerTracer();
+
+  public JakartaServletHttpServerTracer() {
+    super(JakartaServletAccessor.INSTANCE);
+  }
+
+  public static JakartaServletHttpServerTracer tracer() {
+    return TRACER;
+  }
+
+  public Context startSpan(Object servletOrFilter, HttpServletRequest request) {
+    Context context = startSpan(request, getSpanName(servletOrFilter, request, false));
+    // server span name shouldn't be update when server span was created from a call to Servlet
+    // if server span was created from a call to Filter then name may be updated from updateContext.
+    if (servletOrFilter instanceof Servlet) {
+      ServletSpanNaming.setServletUpdatedServerSpanName(context);
+    }
+    return context;
+  }
+
+  private String getSpanName(
+      Object servletOrFilter, HttpServletRequest request, boolean allowNull) {
+    String spanName = getSpanName(servletOrFilter, request);
+    if (spanName == null && !allowNull) {
```

Comment:
I think I only see this called twice, one with allowNull and one without. Can we move the logic for !allowNull to the caller to reduce conditionals?

Reply:
It's good to have as a separate method since `return` keyword is convenient here and the method name still fits. However, made the caller of `true` call the inner method instead to still get rid of this extra parameter, and renamed the inner method to `getSpanNameFromPath`.